### ISSUE TITLE
`GetConnections` should throw an exception if it is called before presynaptic infrastructure is created, but sorting connections is enabled

### ISSUE THREAD

**Issue by jakobj** [jakobj/nest-simulator#28]
By sorting, connection handles would be invalidated. We need to add an additional SLI function that allows the user to construct the presynaptic infrastructure without running Simulate. This would allow the user to run scripts like this:

```postscript
... Connect
InstatiatePresynapticConnectionInfrastructure
<< >> GetConnections
100 Simulate
```

**Comment by jakobj** [comment]
@suku248 have you already worked on this? I remember discussing this before, but don't remember the outcome. If not, could you look into this please?

**Comment by suku248** [comment]
PR #84 addresses this issue.

@heplesser, @stinebuu, @hakonsbm, and I have discussed possible solutions. We decided that updating the connection infrastructure at the beginning of `GetConnections` is the best way to handle this issue.
If `GetConnections` is called before the call to `Simulate`, the connection infrastructure won't be updated again at the beginning of `Simulate` unless connections have been added or removed.
